content/blogposts/2022/sourcegraph-non-devs.md

```diff
@@ -0,0 +1,229 @@
+
+As a product manager at a fast-growing startup, I frequently need to know how our features interact, which analytics metrics other teammates have already built, and which product team processes or standards exist versus those that haven’t been built out yet. I use Sourcegraph to help me quickly answer these types of questions.
+
+### What actions do our metrics track?
+
+I’d just joined Sourcegraph in September 2020 as the first PM for the extensibility team. I went through our Business Intelligence platform but didn’t know what “ViewRegistryExtensionManage” meant or how it was different than “ViewRegistryExtensionContributions.” I turned to Sourcegraph.
+
+**Search:** First, I [ran this search](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+viewRegistryExtensionManage&patternType=regexp), got no results, [ran this search](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+view+RegistryExtensionManage&patternType=regexp), got results, and then [ran this search](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+logViewEvent%28%27RegistryExtension&patternType=literal), which gave me a quick reference for the pages all the events fired on. From there, I could glance through the strings and comments on the page to map them back to states in our app.
+
+<p style={{ width: '100%' }}>
+    <iframe src="https://sourcegraph.com/embed/notebooks/Tm90ZWJvb2s6MjAz?theme=light" frameBorder="0" sandbox="allow-scripts allow-same-origin allow-popups" style={{ width: '100%', height: '300px' }}></iframe>
+</p>
+
+**Impact:** I didn’t need to ask a teammate or pull the repo(s) and search it locally. The latter means I also didn’t have to clutter my desktop with yet another open application window.
+
+**Time Saved:** 10 minutes (20 if we had multiple repos) and I didn’t need another open desktop application.
+
+**Follow-up situation:** We can quickly see the names of all of our view analytics events and be able to go to what page they’re triggered on with [this search](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+logViewEvent%28%27...%27%29+count:all&patternType=structural). (Bonus: with the [Open-in-Figma extension](https://sourcegraph.com/extensions/Joelkw/open-in-figma) I can then 1-click open a mockup of the page to get an immediate understanding of “what page fires this event” without reading the code to figure out what page it renders.)
+
+<p style={{ width: '100%' }}>
+    <iframe src="https://sourcegraph.com/embed/notebooks/Tm90ZWJvb2s6MTc4?theme=light" frameBorder="0" sandbox="allow-scripts allow-same-origin allow-popups" style={{ width: '100%', height: '300px' }}></iframe>
+</p>
+
```

Comment:
```suggestion

### How big a problem are unversioned docs links, and how do I stop people from using them? 

We built a system to version our docs links, but it turns out many people weren't aware of it and were using literal links to docs. (This is bad because it might take someone from an older version to the latest content, which may not be relevant to their version.) 
 
 **Search:** First I ran [this search](https://sourcegraph.com/search?q=context:global+docs.sourcegraph.com+file:client/web/src/enterprise+r:github.com/sourcegraph/sourcegraph%24%403.36&patternType=standard) and found dozens of unversioned docs links. 
 
 <p style={{ width: '100%' }}>
    <iframe src="https://sourcegraph.com/embed/notebooks/ Tm90ZWJvb2s6MTU0Nw==?theme=light" frameBorder="0" sandbox="allow-scripts allow-same-origin allow-popups" style={{ width: '100%', height: '300px' }}></iframe>
</p>

Then, I: 

1. Created a [code monitor](https://docs.sourcegraph.com/code_monitoring) to send off a slack message to our frontend channel whenever someone added another unversioned (bad) link, which looked like this: 

![Code Monitor Docs Alert](https://storage.googleapis.com/sourcegraph-assets/blog/monitor_alert.png) 

2. Migrated a bunch of links to the correct format, tracked them with `to="\/help file:client/web/src/enterprise`, and created a [code insight](https://docs.sourcegraph.com/code_insights) to monitor that migration and ensure it never became a problem again, which looked like this: 

![Code Insights Docs Migration](https://storage.googleapis.com/sourcegraph-assets/blog/Insights_docs_migration.png)

**Impact:** Users don't get lost or confused by landing on the wrong version of a docs page. 

**Time Saved:** 10 minutes I would have spent manually pulling and checking the codebase the first time, plus the many days saved over preventing regression with more difficult or manual automation compared to code monitors and insights.

```